src/workers/newView.ts

```diff
@@ -24,12 +24,104 @@ const addView = async (con: DataSource, entity: View): Promise<boolean> => {
   return false;
 };
 
+const DEFAULT_TIMEZONE = 'UTC'; // in case user doesn't have a timezone set
+const INC_STREAK_QUERY = `
+WITH u AS (
+  SELECT "id", COALESCE("timezone", $3) AS timezone
+  FROM public.user
+  WHERE public.user.id = $1
+),
+vars AS (
+  SELECT (DATE($2 AT TIME ZONE u.timezone)) AS today FROM u
+),
+d AS (
+  SELECT "userId", "currentStreak", today, (today - DATE("lastViewAt" AT TIME ZONE u.timezone)) AS dateDiff, EXTRACT(DOW FROM today) AS dayOfWeek
```

Comment:
`DATE("lastViewAt" AT TIME ZONE u.timezone)` this part simply add a timezone to a non timezone time, or actually move from UTC to the right timezone? If it's the first one I think it's wrong

Reply:
The `lastViewAt` column in `user_streak` table is a `timestamptz` column. So `AT TIME ZONE` gives us the time in the timezone of the user. Postgres always stores times in the timezone that the DB is configured with, but since we give the timezone at the time of writing as well, this should be correct. E.g. imagine a following scenario:

1. User opens an article at 9:00 AM CET. 
2. This is stored as 8:00+00 in Postgres, since Postgres timezone is configured for UTC. 
3. the `"lastViewAt" AT TIME ZONE 'Europe/Vienna'` returns 9:00 AM CET.

So all in all, I think this is what we want. I can add some unit tests for timezones as well, if you think it's worth doing.

Reply:
Yes I agree. This is the expected behavior. Perfect

Reply:
Not 100% sure from the query, but are we storing it under a specific timezone already now?
(think that's what we decided right?) (disregard if i'm wrong here)